torch/fx/graph.py

```diff
@@ -1801,6 +1803,18 @@ def has_side_effect(node):
                 self.erase_node(node)
                 changed = True
 
+        # Call DCE on the subgraphs
+        if self.owning_module is not None:
```

Comment:
when is owning_module set?

Reply:
For some tests (not related to compile), this failed. I think you can have an Fx graph without associating it with a graph module, so this prevents that.